src/lib/modules/Document/DocumentSubjectSearch.js

```diff
@@ -0,0 +1,148 @@
+import { documentApi } from '@api/documents';
+import { Media } from '@components/Media';
+import {
+  setReactSearchKitDefaultSortingOnEmptyQueryString,
+  setReactSearchKitInitialQueryState,
+  setReactSearchKitUrlHandler,
+} from '@config';
+import history from '@history';
+import { SearchControls } from '@modules/SearchControls/SearchControls';
+import { SearchControlsOverridesMap } from '@modules/SearchControls/SearchControlsOverrides';
+import SearchFooter from '@modules/SearchControls/SearchFooter';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OverridableContext } from 'react-overridable';
+import {
+  EmptyResults,
+  Error,
+  InvenioSearchApi,
+  ReactSearchKit,
+  ResultsLoader,
+  ResultsGrid,
+} from 'react-searchkit';
+import { Divider, Container, Ref, Sticky } from 'semantic-ui-react';
+import _isEmpty from 'lodash/isEmpty';
+import { InvenioRequestSerializer } from 'react-searchkit';
+import { DocumentSubjectGrid } from './DocumentSubjectLayout';
+import { Error as IlsError } from '@components/Error';
+import { SearchControlsMobile } from '@modules/SearchControls/SearchControlsMobile';
+
+const getSubjectsQuery = (documentMetadata) => {
+  const subjectSchemes = documentMetadata.subjects.map((subject) => {
+    return `subjects.scheme:"${subject.scheme}"`;
```

Comment:
we need to query not only by scheme but also by values:

<img width="356" alt="Screenshot 2024-03-26 at 09 12 20" src="https://github.com/inveniosoftware/react-invenio-app-ils/assets/38131488/2b8b14d8-2bbb-44d7-b309-319d4cd61eb9">

value + scheme is what defines a subject